### PR TITLE
Investment step 1

### PR DESCRIPTION
**Pull request by jim68000**
Initial version of DHE-663

### CODE REVIEW

File: src/views/investment/index.html [hunk 1]
```diff
@@ -0,0 +1,146 @@
+{% extends "layouts/ukti.html" %}
+
+{% block main %}
+
+{% import "macros/trade.html" as trade %}
+
+<h1 class="heading-xlarge">Create an investment project</h1>
```
Comment (ztolley) [inline src/views/investment/index.html]: Bit like this, though use H1. there's others in there somewhere.

```
<h2 class="page-heading">
  {% if company.id %}<span class="pre-header">Company ref: 321513</span>{% endif %}
  {{ headingName }}
  <span class="post-header">{{ headingAddress }}</span>
</h2>
```

File: src/views/investment/index.html [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+<form class="searchbar" action="/search" method="get" role="search">
+  <div class="searchbar__wrapper">
+    <label class="searchbar__label" for="inv-search">Source of foreign investment</label>
+    <input class="searchbar__input form-control" id="inv-search" type="search" name="term" value="" placeholder="Search for company name or contact">
+    <input class="searchbar__submit" type="submit" value="Search">
+  </div>
+</form>
+
+<div id="inv-results" style="display:none"></div>
```
Comment (ztolley) [inline src/views/investment/index.html]: really... inline styles...

File: src/views/investment/index.html [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    clear: both;
+  }
+
+  .inv-hidden {
+    display: none;
+  }
+
+
+</style>
+<script type="application/javascript">
```
Comment (ztolley) [inline src/views/investment/index.html]: and inline script....

File: src/repositorys/companyrepository.js [hunk 4]
```diff
@@ -301,6 +301,20 @@ function saveCompanyInvestmentSummary (token, summary) {
   return authorisedRequest(token, { url, method, body: data })
 }
 
+function hydrateCompanyInvestments(token, companies) {
+  const promises = []
+  for (const co of companies) {
+    if (co._type !== 'company_companieshousecompany') {
+      promises.push(co) // comapany first
```
Comment (jim68000) [inline src/repositorys/companyrepository.js]: I was surprised this worked, but yes

File: src/lib/countryids.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const ids = {
```
Comment (jim68000) [inline src/lib/countryids.js]: done

File: src/views/investment/index.html (hunk 1 above)
Comment (jim68000) [inline src/views/investment/index.html]: Changed to a class of '.page-heading'

File: src/views/investment/index.html (hunk 3 above)
Comment (jim68000) [inline src/views/investment/index.html]: Done, but now I need to pinch your disclosure stuff

File: src/forms/investment.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
+const searchfield = document.querySelector("#inv-search");
+const resultsdiv = document.querySelector("#inv-results");
+
+const trade = require('@uktrade/trade_elements').elementstuff
+
+const companyTemplate = "<span id='[[conameid]]' class='clickable'>[[name]]</span><span id='[[headclose]]' class='hidden clickable' style='float:right'>Close</span>" +
```
Comment (ztolley) [inline src/forms/investment.js]: es6 strings are a nicer way to do multiline strings and support interpolation

File: src/controllers/investmentcontroller.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+const express = require('express')
+const winston = require('winston')
+const companyRepository = require('../repositorys/companyrepository')
+const metadataRepository = require('../repositorys/metadatarepository')
+const search = require('../services/searchservice')
+
+// imported from company controller
+// @todo merge into utils
+const investmentDetailLabels = {
```
Comment (ztolley) [inline src/controllers/investmentcontroller.js]: split out to own label file in labels folder

File: src/controllers/investmentcontroller.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    .catch((error) => {
+    const errors = error.error
+    if (error.response) {
+      return res.status(error.response.statusCode).json({errors})
+    }
+    next(error)
+  })
+}
+
+function OLDinvsearch(req, res) {
```
Comment (ztolley) [inline src/controllers/investmentcontroller.js]: interesting function name

File: src/sass/application.scss [hunk 9]
```diff
@@ -14,3 +14,25 @@ $images-dir: '/images/';
 @import 'search';
 @import 'shame';
 @import 'splitheader';
+
+
+.inv-result {
```
Comment (ztolley) [inline src/sass/application.scss]: in future put misc styles in the 'shame' file, then if they are really needed create their own or move to trade elements

File: src/controllers/investmentcontroller.js (hunk 8 above)
Comment (jim68000) [inline src/controllers/investmentcontroller.js]: Ah yeah, meant to delete that

File: src/sass/application.scss (hunk 9 above)
Comment (jim68000) [inline src/sass/application.scss]: I wondered what "shame" was for

File: src/controllers/investmentcontroller.js (hunk 7 above)
Comment (jim68000) [inline src/controllers/investmentcontroller.js]: K

File: src/forms/investment.js (hunk 6 above)
Comment (jim68000) [inline src/forms/investment.js]: Yeah, I need to move these over - I originally didn't have them running through webpack - hence the old school approach

Comment (jim68000) [inline src/forms/investment.js]: Done

File: src/controllers/investmentcontroller.js (hunk 8 above)
Comment (jim68000) [inline src/controllers/investmentcontroller.js]: Dpne

File: src/sass/application.scss (hunk 9 above)
Comment (jim68000) [inline src/sass/application.scss]: Done

File: src/controllers/investmentcontroller.js (hunk 7 above)
Comment (jim68000) [inline src/controllers/investmentcontroller.js]: Done

**Pull request merged**